crates/proof-of-sql/src/base/math/non_negative_i32.rs

```diff
@@ -0,0 +1,48 @@
+use alloc::fmt::Display;
+use serde::{Deserialize, Serialize};
+
+/// Type-safe non-negative integer, exists for the sole purpose
+/// of preventing negative values from being used as fixed-size
+/// binary slice widths, due to arrow's unfortunate use of i32 as width.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Hash)]
+pub struct NonNegativeI32(i32);
```

Comment:
We should also bound the size of FixedBinary above by 31. We can't embed 32 bytes canonically inside a scalar.